bridges/snowbridge/pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,210 @@
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use sp_core::H256;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type AssetHubParaId: Get<u32>;
+		type EthereumNetwork: Get<NetworkId>;
+		type WethAddress: Get<H160>;
+		/// XCM message sender
+		type XcmSender: SendXcm;
+		/// To withdraw and deposit an asset.
+		type AssetTransactor: TransactAsset;
+		/// Message relayers are rewarded with this asset
+		type Token: Mutate<Self::AccountId> + Inspect<Self::AccountId>;
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A relayer reward was deposited
+		RewardDeposited {
+			/// The relayer account to which the reward was deposited.
+			account_id: AccountIdOf<T>,
+			/// The reward value.
+			value: BalanceOf<T>,
+		},
+		RewardClaimed {
+			/// The relayer account that claimed the reward.
+			account_id: AccountIdOf<T>,
+			/// The address that received the reward on AH.
+			deposit_address: AccountIdOf<T>,
+			/// The claimed reward value.
+			value: BalanceOf<T>,
+			/// The message ID that was provided, used to track the claim
+			message_id: H256,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// XCMP send failure
+		Send(SendError),
+		/// The relayer rewards balance is lower than the claimed amount.
+		InsufficientFunds,
+		InvalidAmount,
+		InvalidFee,
+	}
+
+	#[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo, PalletError)]
+	pub enum SendError {
+		NotApplicable,
+		NotRoutable,
+		Transport,
+		DestinationUnsupported,
+		ExceedsMaxMessageSize,
+		MissingArgument,
+		Fees,
+	}
+
+	impl<T: Config> From<XcmpSendError> for Error<T> {
+		fn from(e: XcmpSendError) -> Self {
+			match e {
+				XcmpSendError::NotApplicable => Error::<T>::Send(SendError::NotApplicable),
+				XcmpSendError::Unroutable => Error::<T>::Send(SendError::NotRoutable),
+				XcmpSendError::Transport(_) => Error::<T>::Send(SendError::Transport),
+				XcmpSendError::DestinationUnsupported =>
+					Error::<T>::Send(SendError::DestinationUnsupported),
+				XcmpSendError::ExceedsMaxMessageSize =>
+					Error::<T>::Send(SendError::ExceedsMaxMessageSize),
+				XcmpSendError::MissingArgument => Error::<T>::Send(SendError::MissingArgument),
+				XcmpSendError::Fees => Error::<T>::Send(SendError::Fees),
+			}
+		}
+	}
+
+	#[pallet::storage]
+	pub type RewardsMapping<T: Config> =
+		StorageMap<_, Identity, AccountIdOf<T>, BalanceOf<T>, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight((T::WeightInfo::claim(), DispatchClass::Operational))]
+		pub fn claim(
+			origin: OriginFor<T>,
+			deposit_address: AccountIdOf<T>,
```

Comment:
For maximum flexibility, we should allow the funds to be deposited into any account type supported by AH.

```suggestion
			/// Location on AH where funds will be deposited
			deposit_location: Location,
```